surveys/2025/annual-survey/questions.md

```diff
@@ -0,0 +1,918 @@
+- Implementing dynamic library plugins
+- Interoperating with other languages (e.g. C or C++)
+- Achieving structured concurrency with async code
+- Writing executor-agnostic async code
+- Writing correct unsafe code
+- Borrow checker not allowing valid code
+- Slow runtime performance
+- Slow compilation
+- Large binary size of compiled artifacts
+- High disk space usage (e.g. the size of the target folder)
+- Encountering compiler bugs (e.g. ICEs a.k.a. internal compiler errors or miscompilations)
+- Encountering opaque/unclear compiler error messages
+- Subpar IDE support (e.g. some errors are not shown or analysis is slow)
+- Subpar debugging experience (e.g. missing value visualizations or async stacktraces)
+- Lacking documentation of the Rust language or the standard library
+
+Options:
+
+- Big problem for me
+- Could be improved, but does not limit me
+- Not an issue for me at all
+
+> **justification**
+>
+> This question tries to get insights on what people wish the Rust project would improve.
+
+### Are there any challenges not mentioned above that affect your Rust programming productivity?
+
+Type: free form (optional)
+
+> **justification**
+>
+> Allow the cohort to mention other challenges.
+
+### Which features stabilized in the past 12 months do you use the most?
+
+The text in the parentheses shows in which Rust version was the feature stabilized.
+
+Type: matrix (optional)
+
+Features:
+
+- [Strict provenance API](https://blog.rust-lang.org/2025/01/09/Rust-1.84.0/#strict-provenance-apis) (1.84)
+- [Async closures](https://blog.rust-lang.org/2025/02/20/Rust-1.85.0/#async-closures) (1.85)
+- [diagnostic::do_not_recommend](https://blog.rust-lang.org/2025/02/20/Rust-1.85.0/#hiding-trait-implementations-from-diagnostics) (1.85)
+- [Trait upcasting](https://blog.rust-lang.org/2025/04/03/Rust-1.86.0/#trait-upcasting) (1.86)
+- [Anonymous pipes](https://blog.rust-lang.org/2025/05/15/Rust-1.87.0/#anonymous-pipes) (1.87)
+- [Let chains](https://blog.rust-lang.org/2025/06/26/Rust-1.88.0/#let-chains) (1.88)
+- [Naked functions](https://blog.rust-lang.org/2025/06/26/Rust-1.88.0/#naked-functions) (1.88)
+
+Usage:
+
+- I use it or plan to use it
+- I cannot use this feature yet
+- I do not need this feature
+- I did not know it was stabilized
+- I do not know what it is
+
+### How do you build your Rust projects?
+
+Type: select all that apply (optional)
+
+- I use Cargo
+- I use some other build system
+- I combine Cargo and another build system
+- If you use Cargo with (or just use) other build systems, which ones do you use? (open response)
+
+> **justification**
+>
+> cargo team expressed interest in this, see https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Rust.20survery.202023/near/393816653
+
+### From where do you download crates to build Rust projects?
+
+Type: select all that apply (optional)
+
+- crates.io
+- Custom/local/company registry
+- Mirror of crates.io
+- git repositories
+- I don't know
```

Comment:
maybe there should also be an "other" option? e.g. some build systems (CMake) let you depend directly on a .zip file downloaded from a URL at build time.

Reply:
Fair enough, added!